modules/nf-core/pymygene/templates/pymygene.py

```diff
@@ -0,0 +1,215 @@
+        args = parser.parse_args(args_list)
+
+        # Convert "null" values to default values
+        for attr in vars(args):
+            value = getattr(args, attr)
+            if value == "null":
+                setattr(args, attr, parser.get_default(attr))
+
+        # Assign args attributes to self attributes
+        for attr in vars(args):
+            setattr(self, attr, getattr(args, attr))
+
+
+class Version:
+    """
+    This class contains the functions to get the versions of the modules used in the script.
+    """
+
+    @staticmethod
+    def get_versions(modules: list) -> dict:
+        """
+        This function takes a list of modules and returns a dictionary with the versions of each module.
+        This dictionary will also contain the version of python.
+        """
+        dic = {
+            **{"python": '.'.join(map(str, sys.version_info[:3]))},
+            **{module.__name__: module.__version__ for module in modules}
+        }
+        return dic
+
+    @staticmethod
+    def format_yaml_like(data: dict, indent: int = 0) -> str:
+        """Formats a dictionary to a YAML-like string.
+
+        Args:
+            data (dict): The dictionary to format.
+            indent (int): The current indentation level.
+
+        Returns:
+            str: A string formatted as YAML.
+        """
+        yaml_str = ""
+        for key, value in data.items():
+            spaces = "  " * indent
+            if isinstance(value, dict):
+                yaml_str += f"{spaces}{key}:\\n{Version.format_yaml_like(value, indent + 1)}"
+            else:
+                yaml_str += f"{spaces}{key}: {value}\\n"
+        return yaml_str
+
+
+class MyGene:
+    """
+    This class contains the functions to query the MyGene.info API.
+    """
+    def __init__(self, filename, species:str = None, filter_go_category: str = None, filter_go_evidence: str = None) -> None:
+        self.query = self.load_list(filename)
+        self.species = species
+        self.filter_go_category = filter_go_category
+        self.filter_go_evidence = filter_go_evidence
+        self.mg = mygene.MyGeneInfo()
+        self.idmap = self.query2idmap()
+        self.gene_based_info = self.parse_gene_based_info()
+        self.go_based_info = self.parse_go_based_info()
+
+    def load_list(self, filename: str) -> list:
+        """
+        It reads the a list from a file.
+        Each row is an element of the list.
+        """
+        with open(filename, "r") as f:
+            mylist = f.readlines()
+        mylist = [element.strip() for element in mylist]
+        return mylist
+
+    def query2idmap(self) -> dict:
+        """
+        It returns a dictionary with the query ids as keys and the mygene ids as values.
+        """
+        return {dic['_id']: dic['query'] for dic in self.mg.querymany(self.query, species=self.species) if '_id' in dic}
```

Comment:
This is the function you're wrapping, really. So maybe allow a passthrough of unused args to `querymany()`? Might make this more widely useful, just a suggestion.

Reply:
querymany has a lot of arguments though. 
I worry that giving too much flexibility for the user to give any kwords will break the rest of my wrapper script.

Reply:
It's standard practice in nf-core to allow users to pass args through to wrapped software (it's what `task.ext.args` is designed for). I'm not asking you to anticipate all the problems up front, but you shouldn't break that conventions unnecessarily.

Reply:
Check parse_ext_args. There are some optional arguments that are allowed for user to add through task.ext.args.